system/include/compat/emmintrin.h

```diff
@@ -383,8 +383,9 @@ _mm_cvtpd_epi32(__m128d __a)
   int m[2];
   for(int i = 0; i < 2; ++i)
   {
-    int x = lrint(__a[i]);
-    if (x != 0 || fabs(__a[i]) < 2.0)
+    double e = __a[i];
+    int x = lrint(e);
+    if ((x != 0 || fabs(e) < 2.0) && !isinf(e) && e <= LLONG_MAX && e >= LLONG_MIN)
       m[i] = (int)x;
```

Comment:
does it still work if we still use the intrinsic conversion here?

Reply:
unlike in #22893 there is no actual cast here since x is already and `int`.  I've removed the redundant cast to make that more clear.